lib/source/defender/aws_iot_defender_api.c

```diff
@@ -126,17 +168,18 @@ AwsIotDefenderError_t AwsIotDefender_SetMetrics( AwsIotDefenderMetricsGroup_t me
 
 AwsIotDefenderError_t AwsIotDefender_Start( AwsIotDefenderStartInfo_t * pStartInfo )
 {
-    if( pStartInfo == NULL )
-    {
-        IotLogError( "Input start info is invalid." );
+    IOT_FUNCTION_ENTRY( AwsIotDefenderError_t, AWS_IOT_DEFENDER_SUCCESS );
 
-        return AWS_IOT_DEFENDER_INVALID_INPUT;
+    if( ( pStartInfo == NULL ) || ( pStartInfo->mqttConnection == IOT_MQTT_CONNECTION_INITIALIZER ) )
```

Comment:
how about if ```callback``` is not set?

Reply:
callback was optional in earlier API therefore I did not add that check:
https://docs.aws.amazon.com/freertos/latest/lib-ref/c-sdk/defender/structAwsIotDefenderStartInfo__t.html
Since we are changing that, we should make it mandatory.